packages/astro/src/content/types-generator.ts

```diff
@@ -362,6 +362,28 @@ function invalidateVirtualMod(viteServer: ViteDevServer) {
 	viteServer.moduleGraph.invalidateModule(virtualMod);
 }
 
+/**
+ * Takes a configPath and returns a normalized relative version:
+ *  -   If is not relative, it adds `./` to the beginning.
+ *  -   If it ends with `.ts`, it removes the extension.
+ *  -   It adds `.js` to the end.
+ *  -   It stringifies the result (adds `""` around it).
+ * @param from Config path from path.
+ * @param to Config path to path.
+ * @returns Normalized config path.
+ */
+function normalizeConfigPath(from: string, to: string) {
+	const normalizedConfigPath = path.relative(from, to);
+
+	return JSON.stringify(
+		`${isRelativePath(normalizedConfigPath) ? '' : './'}${
+			normalizedConfigPath.endsWith('.ts')
+				? normalizedConfigPath.replace(/\.ts$/, '')
```

Comment:
If this always resolves to a file that ends with `.ts` or `.js` then we can simplify `normalizeConfigPath` a bit by moving that `.js` at the end of the returned string here, so the line ends up being:

```ts
? normalizedConfigPath.replace(/\.ts$/, ".js")
```

Reply:
Fair point. I believe `.mjs` would be possible, and this change would silently leave the `.mjs` extension without changing it

Reply:
I’ll update the PR with the simplified code, then. Thank you!